src/main/java/uk/gov/hmcts/reform/pbis/notify/NotificationClientProvider.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.pbis.notify;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.pbis.ServiceNotFoundException;
+import uk.gov.hmcts.reform.pbis.config.ApplicationConfig;
+import uk.gov.hmcts.reform.pbis.model.EmailTemplateMapping;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientApi;
+
+@Component
+public class NotificationClientProvider {
+
+    private final boolean useClientStub;
```

Comment:
Why not to store the whole config?

Reply:
It's cleaner this way. We don't need to store the config when all we care about is just one flag.